### ISSUE TITLE
Private types export check gets thrown off by mutually recursive types that are exported separately

### ISSUE THREAD

**Issue by rkirov** [Microsoft/TypeScript#14753]
**TypeScript Version:**  2.2.1 
**Code**

```ts
interface A {
  b: B
}

interface B {
  a: A
}
export {A, B}
```

**Expected behavior:**
No errors, all interfaces are exported.

**Actual behavior:**
f.ts(7,6): error TS4033: Property 'b' of exported interface has or is using private name 'B'.

**Comment by rkirov** [comment]
Note simply doing `export interface A` and `export interface B` does not trigger this error.
